### PR TITLE
Translate the startup language select page

### PR DESCRIPTION
**Pull request by shatyuka**
Translate the startup language select page, and language can be changed in real time.
Automatically select the language of current system when starting for the first time.
Change empty string placeholder from `\u2006` (Six-Per-Em Space) to `\u200b` (Zero Width Space).

### CODE REVIEW

**Comment by shatyuka** [comment]
Language tag from https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:39]: This can be simplified to a switch-case statement instead of multiple ifs. Also, you can trim the string to only keep the first 2 or 3 letters.

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:78]: I would personally add a `LogWriteLine`so that we know that the view was refreshed.

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:98]: Same as above

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:106]: You say "fix mouse event", but what does that actually entail? Is it an issue with the library or C#. Please elaborate in your comment.

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:113]: Same as previous.

**Review by Cryotechnic — CHANGES_REQUESTED** [review]
See comments, just some small stuff that needs cleaning up, good otherwise 👍

File: Hi3Helper.Core/Lang/en_US.json
Comment (bagusnl) [inline Hi3Helper.Core/Lang/en_US.json:41]: This was a workaround for how Transifex handles an empty string in a key. If the change is not strictly needed (I forgot how much the gap was with u2008) then I'd not change this because changing base lang will invalidate all localization on that key.

If you think the change is still needed for proper UX, let me know as I will need to be standby to watch Transifex syncing process

File: Hi3Helper.Core/Lang/en_US.json
Comment (bagusnl) [inline Hi3Helper.Core/Lang/en_US.json:46]: Same as above

**Review by bagusnl — COMMENTED** [review]
Just adding a small notes regarding the unicode u2008 change

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (shatyuka) [reply CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:106]: issue with winui3, the dropdown panel of language selection ComboBox collides with non-client area, making the first item not clickable. Use this to disable whole non-client area temporarily. I can move this to MainWindow.

File: Hi3Helper.Core/Lang/en_US.json
Comment (shatyuka) [reply Hi3Helper.Core/Lang/en_US.json:41]: We can keep `\u2006` here, because it's the end of a sentence. Needed for `CDNHelpDetail_1` to avoid a leading white space.

File: Hi3Helper.Core/Lang/en_US.json
Comment (shatyuka) [reply Hi3Helper.Core/Lang/en_US.json:46]: same here

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs
Comment (shatyuka) [reply CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs:39]: Okay. but still need additional process for Chinese and Portuguese, because we have multiple variants in Collapse.

File: CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml
Comment (shatyuka) [inline CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml:72]: Note here: connect all Runs to avoid additional white space. For language that does now need word segmentation like Chinese.

File: Hi3Helper.Core/Lang/en_US.json
Comment (bagusnl) [reply Hi3Helper.Core/Lang/en_US.json:41]: aite, that one is fine since its a new key and wont invalidate any existing localization

**Review by bagusnl — APPROVED** [review]
Tested against main and everything looks great!
Waiting for approval from @Cryotechnic then we can merge by squash

**Review by Cryotechnic — APPROVED** [review]
LGTM, good stuff!

**Pull request merged**
